### ISSUE TITLE
[FEATURE] core: skip 0 wait for timer tasks

### ISSUE THREAD

**Issue by ovidiusas** [OpenSIPS/opensips#3122]
When the processing heavy timer tasks, sometimes those task are overlapping.
For some tasks it is ok to wait an execute them back to back, but for other tasks (most of them) there's no point in re-running the task if already finished.

In timer.c we have compute_wait_with_drift(_tv).
I was running in production with a new type of compute in order to improve efficiency:

#define compute_wait_with_drift_and_skip_0_wait(_tv) \
    do {                                                         \
        if ( drift > ITIMER_TICK ) {                             \
            overruns = drift / uinterval;                        \
            drift = drift % uinterval;                           \
            if(overruns) LM_WARN("Got %u overrun(s) and the drift is %lld\n", overruns, drift);\
            wait = uinterval-drift;                              \
            _tv.tv_sec = wait / 1000000;                         \
            _tv.tv_usec = wait % 1000000;                        \
            drift = 0;                                           \
        } else {                                                 \
            _tv = o_tv;                                          \
        }                                                        \
    }while(0)

Basically, don't re-run the timer task if just got executed. This should improve the overall performance.
There might be some corner cases when we would like to run tasks back to back (I did not found a need for it in my deployments).

**Comment by bogdan-iancu** [comment]
Hi @ovidiusas, thanks for the feedback on this. Here are some clarifications on the matter:
* IMO, it is not up to the timer subsystem to decide if a timer handler is to be run or not. Even with delay or even back2back, there are timer handler which do rely on the fact they are triggered (this of course depends on the logic implemented by the timer) - for example pike, ratelimit module - they to rely on being constantly called in order to monitor the traffic
* the possibility to have a timer skipped if overlapping is already there, but it is not up to the timer subsystem to decide, but up to the timer handler (according to its specifics), see the [TIMER_FLAG_SKIP_ON_DELAY](https://github.com/OpenSIPS/opensips/blob/master/timer.h#L48) flag
